DuckDuckGo/MainViewController.swift

```diff
@@ -1427,10 +1427,14 @@ class MainViewController: UIViewController {
         Task {
             guard case .success(false) = await AccountManager().hasEntitlement(for: .networkProtection) else { return }
 
-            tunnelDefaults.enableEntitlementMessaging()
-
             let controller = NetworkProtectionTunnelController()
+
+            if await controller.isInstalled {
+                tunnelDefaults.enableEntitlementMessaging()
+            }
```

Comment:
We only want to set this messaging if the VPN is installed, otherwise we'll be telling users that the VPN has been disconnected even though it was never present.

There was an issue in the subscription library where it was publishing a notification when changing from no entitlements to an empty array during purchase (subtle difference), so this flag was getting set during that and showing the notification at a confusing time.